README.md

```diff
@@ -1 +1,364 @@
+
+Some other relevant aspects of JSON-B are:
+- If an unrecognized property is found in the JSON data (i.e. it has no matching Java property) it is ignored
+- By default, java fields that did not get matched to incoming JSON data are left as `null`
+- By default, JSON-B only maps to public fields or getter/setter methods. Non-public members are ignored
+- By default, the java objects must have a default (no-args) constructor.
+
+Using this information, try to model the JSON data as POJOs in your app. \
+If you get stuck, there is more information at: http://json-b.net/docs/user-guide.html#default-mapping \
+Or if you really get stuck, the answers can be found here: https://github.com/OpenLiberty/liberty-bikes-ai/tree/checkpoint-3/src/main/java/org/libertybikes/ai/model
+
+
+# 4) Code, Play, Repeat [checkpoint-3]
+
+Now that you've modeled your JSON data with Java objects, you can pass these objects to your `AILogic` and finally make some informed decisions on where to steer your bot! Be warned, there are a few different types of messages that the game-service will broadcast -- not all of them are game ticks! You will need to account for this when passing a `GameTick` to your `AILogic` class somehow:
+
+```java
+    } else {
+        // use jsonb to convert from String --> POJO
+        GameTick gameTick = jsonb.fromJson(message, GameTick.class);
+        if (gameTick.isValid())
+            sendDirection(aiLogic.processAiMove(gameTick));
+    }
+```
+
+The `AILogic` class can house any of the... AI logic... that you want to use to control your bot. Feel free to create additional methods and classes at this point if it would be helpful -- this is essentially the end of the lab and you can do put as much or as little effort as you'd like into making your AI Bot. Perhaps think of some algorithms that you could use to steer your bot. For example, here are a few basic algorithms:
+- Go straight until you are about to run into something, then turn left or right (relative to bot's current direction)
+- At each game tick "look" in each direction -- forward, left, and right. Compute which direction you can go in a straight line for the longest, and go that way.
+
+# 5) Let's Get Ready to Rumble!!!
+
+Towards the end of the lab we will host a competition where everyone can battle their AI bots against each other to see who has the best one! In order to do this, you will run just your AI bot on your laptop. You may now shutdown the core services running on your laptop by running `./gradlew stop` from the root of the liberty-bikes repo.
+
+The lab admins should be running a set of core services on their laptops, and will provde you with their IP address. We will call it `$ADMIN_IP` in these instructions.
+
+In order to configure your bot to connect to the lab admins set of core services, you must do the following:
+1. Go to their frontend at `http://$ADMIN_IP:8080 and register your bot and take note of the generated key like you did at the end of section 2
```

Comment:
Go to their frontend at `http://$ADMIN_IP:8080 -> **port should be 12000**

Reply:
oops, forgot to save the file after editing locally